lib/mongo/server/connection.rb

```diff
@@ -84,11 +89,10 @@ class Connection
       #
       # @since 2.0.0
       def initialize(server, options = {})
-        @address = server.address
         @monitoring = server.monitoring
         @options = options.freeze
         @server = server
-        @ssl_options = options.reject { |k, v| !k.to_s.start_with?(SSL) }
+        @ssl_options = options.reject { |k, v| !k.to_s.start_with?(SSL) }.freeze
```

Comment:
Minor nit: instead of using a double negative of `reject!` and `!`, you can just do `select!`

Reply:
Changed.